File: CS/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraCharts;

namespace WindowsFormsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            this.ordersTableAdapter.Fill(this.nwindDataSet.Orders);

            this.comboBox1.SelectedIndex = 0;
        }

        private void comboBox1_SelectedIndexChanged(object sender, EventArgs e) {
            AxisX axis = ((XYDiagram)this.chartControl1.Diagram).AxisX;

            switch (comboBox1.SelectedIndex) {
                case 0:
                    axis.DateTimeScaleMode = DateTimeScaleMode.Manual;
                    axis.DateTimeMeasureUnit = DateTimeMeasurementUnit.Day;
                    axis.DateTimeGridAlignment = DateTimeMeasurementUnit.Day;
                    break;
                case 1:
                    axis.DateTimeScaleMode = DateTimeScaleMode.AutomaticAverage;
                    break;
                case 2:
                    axis.DateTimeScaleMode = DateTimeScaleMode.AutomaticIntegral;
                    break;
            }
        }

    }
}

File: VB/Form1.vb
﻿Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraCharts


Public Class Form1

    Private Sub Form1_Load(sender As System.Object, e As System.EventArgs) _
    Handles MyBase.Load
        Me.OrdersTableAdapter.Fill(Me.NwindDataSet.Orders)

        Me.comboBox1.SelectedIndex = 0
    End Sub

    Private Sub comboBox1_SelectedIndexChanged(sender As System.Object, e As System.EventArgs) _
    Handles comboBox1.SelectedIndexChanged
        Dim axis As AxisX
        axis = CType(Me.ChartControl1.Diagram, XYDiagram).AxisX

        Select Case comboBox1.SelectedIndex
            Case 0
                axis.DateTimeScaleMode = DateTimeScaleMode.Manual
                axis.DateTimeMeasureUnit = DateTimeMeasurementUnit.Day
                axis.DateTimeGridAlignment = DateTimeMeasurementUnit.Day
            Case 1
                axis.DateTimeScaleMode = DateTimeScaleMode.AutomaticAverage
            Case 2
                axis.DateTimeScaleMode = DateTimeScaleMode.AutomaticIntegral
        End Select
    End Sub
End Class